File: 2021/day01/go/part1/part1.go
package part1

import "strconv"

func Parse(lines []string) []int {
	var values []int
	for _, line := range lines {
		value, err := strconv.Atoi(line)
		if err != nil {
			panic(err)
		}
		values = append(values, value)
	}
	return values
}

func ComputeAnswer(lines []string) int {
	result := 0
	depths := Parse(lines)
	for index := 1; index < len(depths); index++ {
		if depths[index] > depths[index-1] {
			result += 1
		}
	}
	return result
}

File: 2021/day01/python/part1.py
from typing import List


def parse(lines: List[str]) -> List[int]:
    return [int(_) for _ in lines]


def compute_answer(lines: List[str]) -> int:
    result = 0
    depths = parse(lines)
    for index in range(1, len(depths)):
        if depths[index] > depths[index - 1]:
            result += 1
    return result
